mimic/rest/auth_api.py

```diff
@@ -21,24 +23,86 @@
 from mimic.session import NonMatchingTenantError
 from mimic.util.helper import invalid_resource
 
+from mimic.model.behaviors import BehaviorRegistry, EventDescription
+
 Request.defaultContentType = 'application/json'
 
 
-class AuthApi(object):
+authentication = EventDescription()
+"""
+Event refers to authenticating against Identity using a username/password,
+username/api-key, token, or getting an impersonation token.
+"""
 
+
+@authentication.declare_default_behavior
+def default_authentication_behavior(core, http_request, credentials):
+    """
+    Default behavior in response to a server creation.
+
+    :param core: An instance of :class:`mimic.core.MimicCore`
+    :param http_request: A twisted http request/response object
+    :param credentials: An `mimic.model.identity.ICredentials` provider
```

Comment:
Given that this is the main documentation of what the signature for the `authentication` event is, it should really also include a `:return:` field as well.